DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -592,7 +592,28 @@ public DMASTProcStatement ProcStatement()
                 }
 
                 return varDeclarations.ToArray();
-            } else if (hasNewline) {
+            } else if (Check(TokenType.DM_LeftCurlyBracket)) {
+                Whitespace();
+                Newline();
+                bool isIndented = Check(TokenType.DM_Indent);
+
+                List<DMASTProcStatementVarDeclaration> varDeclarations = new();
+                TokenType type = isIndented ? TokenType.DM_Dedent : TokenType.DM_RightCurlyBracket;
+                while (!Check(type)) {
+                    DMASTProcStatementVarDeclaration[] varDecl = ProcVarEnd(true, path: varPath);
+                    Check(TokenType.DM_Semicolon);
+                    if (varDecl == null) Error("Expected a var declaration");
+
+                    varDeclarations.AddRange(varDecl);
+                    Whitespace();
+                    Newline();
+                }
+
+                if (isIndented) Consume(TokenType.DM_RightCurlyBracket, "Expected '}'");
```

Comment:
A newline comes before `}` tokens.
```suggestion
                if (isIndented) {
                    Newline();
                    Consume(TokenType.DM_RightCurlyBracket, "Expected '}'");
                }
```